### PR TITLE
MPGD Hits for new Tracking Config

### PR DESCRIPTION
**Pull request by mposik1983**
### Briefly, what does this PR introduce?
Add MPGD Hits to new MPGD layers (outer barrel, forward and backward endcaps)

### What kind of change does this PR introduce?
- [ ] Bug fix (issue #__)
- [ ] New feature (issue #__)
- [ ] Documentation update
- [x] Other: Adds hits for new tracking geometry

### Please check if this PR fulfills the following:
- [ ] Tests for the changes have been added
- [ ] Documentation has been added / updated
- [x] Changes have been communicated to collaborators

### Does this PR introduce breaking changes? What changes might users need to make to their code?
No.

### Does this PR change default behavior?
No

### CODE REVIEW

**Comment by wdconinc** [comment]
We should use RawHits instead of DigiHits. I hadn't noticed that tracking uses something against our naming conventions (which are based on the type name in EDM4hep and EDM4eic).

**Comment by mposik1983** [comment]
> We should use RawHits instead of DigiHits. I hadn't noticed that tracking uses something against our naming conventions (which are based on the type name in EDM4hep and EDM4eic).

I see. I was just updating the existing code. I will change "Digi" to "Raw" then in all tracking cases

File: src/detectors/MPGD/MPGD.cc [hunk 1]
```diff
@@ -23,30 +23,72 @@ void InitPlugin(JApplication *app) {
...
             "MPGDBarrelRecHits",    // Output data tag
              tracker_hit_reco_cfg));           // Hit reco default config for factories
 
     // Digitization
     SiliconTrackerDigiConfig dirc_digi_default_cfg;
-    barrel_digi_default_cfg.threshold = 0.25 * dd4hep::keV;
-    barrel_digi_default_cfg.timeResolution = 8;
-    app->Add(new JChainFactoryGeneratorT<SiliconTrackerDigi_factory>({"MPGDDIRCHits"}, "MPGDDIRCDigiHits", dirc_digi_default_cfg));
+    dirc_digi_default_cfg.threshold = 0.25 * dd4hep::keV;
+    dirc_digi_default_cfg.timeResolution = 10;
```
Comment (wdconinc) [inline src/detectors/MPGD/MPGD.cc:40]: This stupid typo mistake could/should have been avoided with a structure like this:
```
    SiliconTrackerDigiConfig dirc_digi_default_cfg {
      .threshold = 0.25 * dd4hep::keV,
      .timeResolution = 10,
    };
```

Comment (wdconinc) [reply src/detectors/MPGD/MPGD.cc:40]: (not your typo, but the pre-existing typo you fixed)

File: src/detectors/MPGD/MPGD.cc [hunk 2]
```diff
@@ -23,30 +23,72 @@ void InitPlugin(JApplication *app) {
...
     app->Add(new JChainFactoryGeneratorT<TrackerHitReconstruction_factory>(
-            {"MPGDDIRCDigiHits"},     // Input data collection tags
+            {"MPGDDIRCRawHits"},     // Input data collection tags
             "MPGDDIRCRecHits",   // Output data tag
             dirc_hit_reco_cfg));          // Hit reco default config for factories
 
+    // Digitization
+    SiliconTrackerDigiConfig outerMPGDbarrel_digi_default_cfg;
+   outerMPGDbarrel_digi_default_cfg.threshold = 0.25 * dd4hep::keV;
+   outerMPGDbarrel_digi_default_cfg.timeResolution = 10;
```
Comment (wdconinc) [inline src/detectors/MPGD/MPGD.cc:54]:
```suggestion
    outerMPGDbarrel_digi_default_cfg.threshold = 0.25 * dd4hep::keV;
    outerMPGDbarrel_digi_default_cfg.timeResolution = 10;
```

File: src/detectors/MPGD/MPGD.cc (hunk 1 above)
Comment (wdconinc) [reply src/detectors/MPGD/MPGD.cc:40]: Let's postpone this to a different PR.

**Review by wdconinc — DISMISSED** [review]

File: src/detectors/MPGD/MPGD.cc (hunk 1 above)
Comment (mposik1983) [reply src/detectors/MPGD/MPGD.cc:40]: I do like that structure, it is much easier to read!

Comment (wdconinc) [reply src/detectors/MPGD/MPGD.cc:40]: See https://github.com/eic/EICrecon/issues/746

Also, I think the pythonic trailing comma on the `timeResolution` line may not work (or maybe it does... I think this is the one place where C(++) may actually allow trailing commas.)

**Review by wdconinc — APPROVED** [review]

**Comment by veprbl** [comment]
This broke the tracking (messages `CKFTracking    WARNING   No tracks found`)

**Pull request merged**